### PR TITLE
be:spdk: add support for subnqn

### PR DESCRIPTION
**Pull request by karlowich**
Make it possible to specify subnqn when running `xnvme enum` or `xnvme info` using NVMe over Fabrics.

Before this can be merged a test case for this must be added:
- [x] add test cases for subnqn

### CODE REVIEW

File: lib/xnvme_be_spdk_dev.c [hunk 1]
```diff
@@ -683,9 +683,10 @@ xnvme_be_spdk_enumerate(const char *sys_uri, struct xnvme_opts *opts, xnvme_enum
 			}
 
 			snprintf(trid_str, sizeof trid_str,
-				 "trtype:%s adrfam:%s traddr:%s trsvcid:%d",
+				 "trtype:%s adrfam:%s traddr:%s trsvcid:%d subnqn:'%s'",
```
[the remark this answers is not in the document]
Comment (karlowich) [reply lib/xnvme_be_spdk_dev.c]: Thank you for your feedback!
I have removed the ticks.

As for line 696: You're right I don't know how I missed that. 
Unless that line is changed this PR is useless...
I wonder why we always set the subnqn to 'nqn.2014-08.org.nvmexpress.discovery' 
@safl What do you think?

**Comment by karlowich** [comment]
@safl The failing test is fixed by #126, it is due to the `cref`/`deref` stuff

**Comment by karlowich** [comment]
[05a547f](https://github.com/OpenMPDK/xNVMe/pull/110/commits/05a547f661794618808c1e22bec1949af9321d9a) Introduces a qemu setup with two controllers. This causes `xnvme_tests_enum open --count 4 --be spdk` to fail.
I have applied [2b5d955](https://github.com/OpenMPDK/xNVMe/pull/110/commits/2b5d95574ac56c888fb1a13a3b12ab04f5c921d0) to fix this. Thanks @LiadOz

File: lib/xnvmec.c [hunk 2]
```diff
@@ -468,6 +468,11 @@ static struct xnvmec_opt_attr xnvmec_opts[] = {
 		.name = "uri",
 		.descr = "System URI e.g. '10.9.8.1:8888'",
 	},
+	{.opt = XNVMEC_OPT_SUBNQN,
```
Comment (safl) [inline lib/xnvmec.c:471]: Seems like the clang-formater did not catch this. Perhaps add a newline?

**Review by safl — CHANGES_REQUESTED** [review]
LGTM, minor stuff:

* ``be:spdk: add opt for subnqn`` to ``api,xnvmec,be:spdk: add subnqn``
* ``ci: test fabrics with subnqn`` to ``ci,toolbox:cijoe: test fabrics with subnqn``
And a couple of inline comments.
Thanks!

**Review by safl — APPROVED** [review]

**Pull request merged**